promptsource/augmentation.py

```diff
@@ -0,0 +1,86 @@
+"""A Draft of data augmentation for prompt/output instance.
+
+This simple and stupid draft is meant to be a pot of stone soup.
+
+Example:
+    `augment("Business")` returns `"Enterprises"`
+
+Note:
+    The back-translation strategy uses two different models for increasing the diversity.
+    The risk of distortion is as the above example shown.
+
+Todo:
+    * Speed and memory-footprint
+    * Refactoring for a better implementation of Strategy pattern
+    * API for batch processing
+    * Special tokens that indicate augmentation invariants
+"""
+import re
+from typing import Iterable, Mapping
+
+from jinja2 import Environment
+from transformers import AutoModelForSeq2SeqLM, AutoTokenizer, pipeline
+
+
+jinja_env = Environment()
+en_fr_translator = pipeline("translation_en_to_fr")
+fr_en_model_name = "Helsinki-NLP/opus-mt-ROMANCE-en"
+fr_en_translator = pipeline(
+    model=AutoModelForSeq2SeqLM.from_pretrained(fr_en_model_name),
+    tokenizer=AutoTokenizer.from_pretrained(fr_en_model_name),
+    task="translation_fr_to_en",
+)
+
+CONST_REGEX = re.compile(r"(?P<cnst>{{\"[^\"]*\"}})")
+
+
+def _decorate_constants(tmpl_rows: Iterable[str]) -> (Iterable[str], Mapping[str, str]):
```

Comment:
I failed to find a way of controlling Jinja constants so this is based on regex...